packages/core/chrome/core-chrome-browser-internal/src/ui/project/header.tsx

```diff
@@ -271,20 +272,12 @@ export const ProjectHeader = ({
             </EuiHeaderSection>
           </EuiHeader>
 
-          <EuiHeader
-            position="fixed"
-            className="header__secondBar"
-            data-test-subj="kibanaProjectHeaderActionMenu"
-          >
-            <EuiHeaderSection />
-            {headerActionMenuMounter.mount && (
-              <EuiHeaderSection side="right">
-                <EuiHeaderSectionItem>
-                  <HeaderActionMenu mounter={headerActionMenuMounter} />
-                </EuiHeaderSectionItem>
-              </EuiHeaderSection>
-            )}
-          </EuiHeader>
+          {headerActionMenuMounter.mount && (
+            <AppMenuBar
+              isOpen={isOpen ?? false}
+              headerActionMenuMounter={headerActionMenuMounter}
+            />
+          )}
```

Comment:
This is not really feasible, without finding a way to send the `actionMenu$` prop into the app content area itself and giving control of it to the app, which adds a risk of inconsistency.

I understand the concern that the app menu bar is not semantically part of the header. This could have accessibility repercussions if for some reason we need to avoid browsers from thinking it is.

From my perspective, this is not a new architecture, and is one that my team has inherited from the core team, and I'm hesitant to change the architecture in this PR. Perhaps the concern can be tracked as a backlog issue?

Reply:
@MichaelMarcialis an update: Cee worked with me to greatly smooth out the styles for positioning the app menu bar with the `position: sticky` rule. This led to us moving the markup for the menu bar outside of the `header` element. It's not quite as close to the content as you have diagrammed, but it does hopefully solve the issue with semantics!